src/components/colorscale/has_colorscale.js

```diff
@@ -6,24 +6,22 @@
 * LICENSE file in the root directory of this source tree.
 */
 
-
 'use strict';
 
 var isNumeric = require('fast-isnumeric');
-
 var Lib = require('../../lib');
-
 var isValidScale = require('./is_valid_scale');
 
-
 module.exports = function hasColorscale(trace, containerStr) {
     var container = containerStr ?
             Lib.nestedProperty(trace, containerStr).get() || {} :
             trace,
         color = container.color,
         isArrayWithOneNumber = false;
 
-    if(Array.isArray(color)) {
+    if(Lib.isTypedArray(color)) {
+        isArrayWithOneNumber = true;
+    } else if(Array.isArray(color)) {
```

Comment:
again to my question about missing data in typed arrays... here can't we just let these typed arrays drop into the for loop as well?

Reply:
done and :lock: in 306986d83c76d65ddc7a8edd0d463d86ba7d0db5